optuna/_gp/_search_space.py

```diff
@@ -0,0 +1,162 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from enum import IntEnum
+import math
+from typing import Any
+
+import numpy as np
+import scipy.stats.qmc
+
+import optuna
```

Comment:
What about importing them to reduce lines?
```suggestion
from optuna.distributions import BaseDistribution
from optuna.distributions import CategoricalDistribution
from optuna.distributions import FloatDistribution
from optuna.distributions import IntDistribution
from optuna.trial import FrozenTrial
```